pkg/bootstrap/platform/gcp.go

```diff
@@ -122,7 +123,23 @@ func IsGCP() bool {
 		// Assume this is running on GCP if GCP project env variable is set.
 		return true
 	}
-	return metadata.OnGCE()
+
+	timeC := time.After(time.Millisecond * 100)
+	ch := make(chan bool)
+	go func() {
+		ch <- metadata.OnGCE()
```

Comment:
This can be blocked when real timeout happen

Reply:
Just FYI. In k8s, it just check the local file `/sys/class/dmi/id/product_name` to decide whether it is on GCE or not
https://github.com/kubernetes/kubernetes/blob/master/pkg/credentialprovider/gcp/metadata.go#L137

Reply:
ok, I will revisit that, thanks.